tooling/prometheus-rules/main.go

```diff
@@ -45,10 +45,6 @@ func (o *options) complete(args []string) error {
 	if err := yaml.Unmarshal(rawRules, &o.rules); err != nil {
 		return fmt.Errorf("failed to parse input rules: %v", err)
 	}
-	o.output, err = os.Create(o.outputBicep)
```

Comment:
Why did this move out of `complete`? The point of this pattern is to have all necessary data read in and created here so testing past this step is easy.

Reply:
Oh, my point was, why create the output if the test could fail?

Reply:
Fair, but losing out on having options as useful input to execution is a shame.